common-messaging/src/main/scala/org/genivi/sota/messaging/daemon/MessageBusListener.scala

```diff
@@ -0,0 +1,72 @@
+package org.genivi.sota.messaging.daemon
+
+import akka.NotUsed
+import akka.actor.{Actor, ActorLogging, ActorRef, Props, Terminated}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, Source}
+import cats.data.Xor
+import com.typesafe.config.ConfigException
+import org.genivi.sota.messaging.MessageBus
+import org.genivi.sota.messaging.Messages.{Message, MessageLike}
+import org.genivi.sota.messaging.daemon.MessageBusListenerActor.Subscribe
+import scala.concurrent.duration._
+import scala.util.Try
+
+class MessageBusListenerActor[Msg <: Message](streamName: String,
+                                              props: Props,
+                                              mlArg: MessageLike[Msg]) extends Actor with ActorLogging {
+
+  implicit val materializer = ActorMaterializer()
+  implicit val ec = context.dispatcher
+  implicit val ml = mlArg
+
+  override def postRestart(reason: Throwable): Unit = trySubscribeDelayed()
+
+  private def subscribed(listener: ActorRef): Receive = {
+    context watch listener
+
+    {
+      case Terminated(_) =>
+        log.error("Source/Listener died, subscribing again")
+        trySubscribeDelayed()
+        context become idle
+    }
+  }
+
+
+  private def subscribe(): Unit = {
+    log.info(s"Subscribing to ${streamName}")
```

Comment:
Also, I think this can just be `mlArg.streamName`, then you won't have to pass in the stream name explicitly.